tools/make/kube.mk

```diff
@@ -123,16 +123,20 @@ run-e2e: install-e2e-telemetry
 	kubectl apply -f test/config/gatewayclass.yaml
 ifeq ($(E2E_RUN_TEST),)
 	go test -v -tags e2e ./test/e2e --gateway-class=envoy-gateway --debug=true --cleanup-base-resources=false
-	go test -v -tags e2e ./test/e2e/upgrade --gateway-class=upgrade --debug=true --cleanup-base-resources=$(E2E_CLEANUP)
+	go test -v -tags e2e ./test/e2e/upgrade --gateway-class=upgrade --debug=true --cleanup-base-resources=false
```

Comment:
if there's a really matter, you can split to into a single make target(e.g. `make e2e-merge-gateways`)
and I don't think we should run it on every k8s version, maybe just the latest.

Reply:
@zirain we run all tests on all versions, we should probably continue to do the same here

Reply:
maybe recreate the cluster, that would be faster?

Reply:
The upgrade test relys on the base resources that created by the first e2e test, since the merge gateways test also relys on this base resources, we may not want to cleanup after this test finished. So only cleanup these base resources for the last test. 

In terms of recreating cluster, logically speaking, this could not be the fast way. If we recreate the cluster, that will cause base resources to setup again.